### ISSUE TITLE
Field Names that look alike are annotated wrong?

### ISSUE THREAD

**Issue by axyr** [silverleague/silverstripe-ideannotator#68]
``` php
<?php


/**
 * Class CustomerSessionPage
 *
 * @property string $EmailFrom
 * @property string $EmailFromName
 * @property string $EmailSubject
 * @property string $EmailContentTitle <=========
 * @property string $EmailContentTitle <=========
 * @property string $ThankYouContent
 * @property string $SubmitButtonText
 * @method DataList|CustomerSession[] CustomerSessions()
 * @method DataList|CustomerSessionSubmission[] CustomerSessionSubmissions()
 */
class CustomerSessionPage extends Page
{
    private static $db = array(
        'EmailFrom'         => 'Varchar(255)',
        'EmailFromName'     => 'Varchar(255)',
        'EmailSubject'      => 'Varchar(255)',
        'EmailContentTitle' => 'Varchar(255)', <=========
        'EmailContent'      => 'Text', <=========
        'ThankYouContent'   => 'HTMLText',
        'SubmitButtonText'  => 'Varchar(255)',
    );
}
```

**Comment by axyr** [comment]
Not able to reproduce this... closed for now...

**Comment by Firesphere** [comment]
I have just reproduced this one. Same one, longer one, followed by the shorter one breaks.

**Comment by Firesphere** [comment]
My code generating the error:
```
private static $db = array(
    'StatusCodeDescription' => 'Varchar(50)'
    'StatusCode' => 'Int',
);
```
It's a weird one. Once it is wrong, it seems to stay wrong, but after removing the faulty declarations, it seems to somehow magically work fine again.
Remove the entire class, build and re-add the class, and it seems to go wrong again.
I have not thoroughly tested this due to lack of time. But it is happening.

**Comment by Firesphere** [comment]
It seems, this is caused by a string-matching. If the string matches initially, it goes wrong. So "longDescription" followed by "long", will match as being the same, since the first starts with the same string as the second.
If I make the second one "clong", the problem does not arise.
For now, I'd say, make sure the shorter version is above the longer one. Haven't found a solution besides clearing the array, which obviously causes other issues, like nothing being written to the file.

**Comment by cristiandeschamps** [comment]
@Firesphere thanks! that's just resolved my problem!

**Comment by Firesphere** [comment]
Ping myself, investigate what's going on here

**Comment by Firesphere** [comment]
Solved in #107 for SS4
